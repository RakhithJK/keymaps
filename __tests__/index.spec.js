import keymap, { generateKeymap } from '../index';
import mockGlobal from './__mocks__/mockGlobal.json';
import mockRegion1 from './__mocks__/mockRegion1.json';
import mockRegion2 from './__mocks__/mockRegion2.json';

describe('keymap', () => {
  test('should output the custom keymap', () => {
    expect(
      generateKeymap(mockGlobal, [mockRegion1, mockRegion2]),
    ).toMatchSnapshot();
  });

  test('should output the entire keymap', () => {
    expect(keymap).toMatchSnapshot();
  });
});
