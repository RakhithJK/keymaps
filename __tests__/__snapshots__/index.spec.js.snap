// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keymap should output the custom keymap 1`] = `
Object {
  "humanReadable": Object {
    "global": Object {
      "keys": Object {
        "ENTER": "28",
      },
      "language": "mockedGlobal",
      "modifiers": Object {
        "MOD_LSHIFT": "02",
      },
    },
    "regions": Object {
      "mockedRegion1": Object {
        "keys": Object {
          "A": "02,04",
          "a": "00,04",
        },
        "language": "mockedRegion1",
      },
      "mockedRegion2": Object {
        "keys": Object {
          "7": "02,24",
          "è": "00,24",
        },
        "language": "mockedRegion2",
      },
    },
  },
  "scancodes": Object {
    "global": Object {
      "keys": Object {
        "28": "ENTER",
      },
      "language": "mockedGlobal",
      "modifiers": Object {
        "02": "MOD_LSHIFT",
      },
    },
    "regions": Object {
      "mockedRegion1": Object {
        "keys": Object {
          "04": Array [
            "a",
            "A",
          ],
        },
        "language": "mockedRegion1",
      },
      "mockedRegion2": Object {
        "keys": Object {
          "24": Array [
            "è",
            "7",
          ],
        },
        "language": "mockedRegion2",
      },
    },
  },
}
`;

exports[`keymap should output the entire keymap 1`] = `
Object {
  "humanReadable": Object {
    "global": Object {
      "keys": Object {
        " ": "2c",
        "BACKSPACE": "2a",
        "ENTER": "28",
        "ESC": "29",
        "KEY_102ND": "64",
        "KEY_AGAIN": "79",
        "KEY_ALTERASE": "99",
        "KEY_CANCEL": "9b",
        "KEY_CAPSLOCK": "39",
        "KEY_CLEAR": "a2",
        "KEY_COMPOSE": "65",
        "KEY_COPY": "7c",
        "KEY_CRSEL": "a3",
        "KEY_CUT": "7b",
        "KEY_DELETE": "4c",
        "KEY_DOWNARROW": "51",
        "KEY_END": "4d",
        "KEY_EXSEL": "a4",
        "KEY_F1": "3a",
        "KEY_F10": "43",
        "KEY_F11": "44",
        "KEY_F12": "45",
        "KEY_F13": "68",
        "KEY_F14": "69",
        "KEY_F15": "6a",
        "KEY_F16": "6b",
        "KEY_F17": "6c",
        "KEY_F18": "6d",
        "KEY_F19": "6e",
        "KEY_F2": "3b",
        "KEY_F20": "6f",
        "KEY_F21": "70",
        "KEY_F22": "71",
        "KEY_F23": "72",
        "KEY_F24": "73",
        "KEY_F3": "3c",
        "KEY_F4": "3d",
        "KEY_F5": "3e",
        "KEY_F6": "3f",
        "KEY_F7": "40",
        "KEY_F8": "41",
        "KEY_F9": "42",
        "KEY_FIND": "7e",
        "KEY_FRONT": "77",
        "KEY_HANGEUL": "90",
        "KEY_HANJA": "91",
        "KEY_HELP": "75",
        "KEY_HENKAN": "8a",
        "KEY_HIRAGANA": "93",
        "KEY_HOME": "4a",
        "KEY_INSERT": "49",
        "KEY_INT7": "8d",
        "KEY_INT8": "8e",
        "KEY_INT9": "8f",
        "KEY_KATAKANA": "92",
        "KEY_KATAKANAHIRAGANA": "88",
        "KEY_KPAD_!": "cf",
        "KEY_KPAD_#": "cc",
        "KEY_KPAD_%": "c4",
        "KEY_KPAD_&": "c7",
        "KEY_KPAD_&&": "c8",
        "KEY_KPAD_0": "62",
        "KEY_KPAD_00": "b0",
        "KEY_KPAD_000": "b1",
        "KEY_KPAD_1": "59",
        "KEY_KPAD_2": "5a",
        "KEY_KPAD_3": "5b",
        "KEY_KPAD_4": "5c",
        "KEY_KPAD_5": "5d",
        "KEY_KPAD_6": "5e",
        "KEY_KPAD_7": "5f",
        "KEY_KPAD_8": "60",
        "KEY_KPAD_9": "61",
        "KEY_KPAD_:": "cb",
        "KEY_KPAD_<": "c5",
        "KEY_KPAD_>": "c6",
        "KEY_KPAD_@": "ce",
        "KEY_KPAD_A": "bc",
        "KEY_KPAD_ASTERISK": "55",
        "KEY_KPAD_B": "bd",
        "KEY_KPAD_BINARY": "da",
        "KEY_KPAD_Backspace": "bb",
        "KEY_KPAD_C": "be",
        "KEY_KPAD_CLEAR": "d8",
        "KEY_KPAD_CLEARENTRY": "d9",
        "KEY_KPAD_COMMA": "85",
        "KEY_KPAD_CURRENCYSUBUNIT": "b5",
        "KEY_KPAD_CURRENCYUNIT": "b4",
        "KEY_KPAD_D": "bf",
        "KEY_KPAD_DECIMAL": "dc",
        "KEY_KPAD_DECIMALSEPARATOR": "b3",
        "KEY_KPAD_DOT": "63",
        "KEY_KPAD_E": "c0",
        "KEY_KPAD_ENTER": "58",
        "KEY_KPAD_EQUAL": "86",
        "KEY_KPAD_F": "c1",
        "KEY_KPAD_HEXADECIMAL": "dd",
        "KEY_KPAD_LEFTPAREN": "b6",
        "KEY_KPAD_MEMADD": "d3",
        "KEY_KPAD_MEMCLEAR": "d2",
        "KEY_KPAD_MEMDIV": "d6",
        "KEY_KPAD_MEMMULT": "d5",
        "KEY_KPAD_MEMRECALL": "d1",
        "KEY_KPAD_MEMSTORE": "d0",
        "KEY_KPAD_MEMSUB": "d4",
        "KEY_KPAD_MINUS": "56",
        "KEY_KPAD_OCTAL": "db",
        "KEY_KPAD_PLUS": "57",
        "KEY_KPAD_PLUSMINUS": "d7",
        "KEY_KPAD_RIGHTPAREN": "b7",
        "KEY_KPAD_SLASH": "54",
        "KEY_KPAD_Space": "cd",
        "KEY_KPAD_THOUSANDSSEPARATOR": "b2",
        "KEY_KPAD_Tab": "ba",
        "KEY_KPAD_XOR": "c2",
        "KEY_KPAD_^": "c3",
        "KEY_KPAD_{": "b8",
        "KEY_KPAD_|": "c9",
        "KEY_KPAD_||": "ca",
        "KEY_KPAD_}": "b9",
        "KEY_KPEQUAL": "67",
        "KEY_KPJPCOMMA": "8c",
        "KEY_LANG6": "95",
        "KEY_LANG7": "96",
        "KEY_LANG8": "97",
        "KEY_LANG9": "98",
        "KEY_LEFTALT": "e2",
        "KEY_LEFTARROW": "50",
        "KEY_LEFTCTRL": "e0",
        "KEY_LEFTMETA": "e3",
        "KEY_LEFTSHIFT": "e1",
        "KEY_LOCKING_CAPSLOCK": "82",
        "KEY_LOCKING_NUMLOCK": "83",
        "KEY_LOCKING_SCROLLLOCK": "84",
        "KEY_MEDIA_BACK": "f1",
        "KEY_MEDIA_CALC": "fb",
        "KEY_MEDIA_COFFEE": "f9",
        "KEY_MEDIA_EDIT": "f7",
        "KEY_MEDIA_EJECTCD": "ec",
        "KEY_MEDIA_FIND": "f4",
        "KEY_MEDIA_FORWARD": "f2",
        "KEY_MEDIA_MUTE": "ef",
        "KEY_MEDIA_NEXTSONG": "eb",
        "KEY_MEDIA_PLAYPAUSE": "e8",
        "KEY_MEDIA_PREVIOUSSONG": "ea",
        "KEY_MEDIA_REFRESH": "fa",
        "KEY_MEDIA_SCROLLDOWN": "f6",
        "KEY_MEDIA_SCROLLUP": "f5",
        "KEY_MEDIA_SLEEP": "f8",
        "KEY_MEDIA_STOP": "f3",
        "KEY_MEDIA_STOPCD": "e9",
        "KEY_MEDIA_VOLUMEDOWN": "ee",
        "KEY_MEDIA_VOLUMEUP": "ed",
        "KEY_MEDIA_WWW": "f0",
        "KEY_MUHENKAN": "8b",
        "KEY_MUTE": "7f",
        "KEY_NUMLOCK": "53",
        "KEY_OPEN": "74",
        "KEY_OPER": "a1",
        "KEY_OUT": "a0",
        "KEY_PAGEDOWN": "4e",
        "KEY_PAGEUP": "4b",
        "KEY_PASTE": "7d",
        "KEY_PAUSE": "48",
        "KEY_POWER": "66",
        "KEY_PRINTSCREEN": "46",
        "KEY_PRIOR": "9d",
        "KEY_PROPS": "76",
        "KEY_RETURN": "9e",
        "KEY_RIGHTALT": "e6",
        "KEY_RIGHTARROW": "4f",
        "KEY_RIGHTCTRL": "e4",
        "KEY_RIGHTMETA": "e7",
        "KEY_RIGHTSHIFT": "e5",
        "KEY_RO": "87",
        "KEY_SCROLLLOCK": "47",
        "KEY_SEPARATOR": "9f",
        "KEY_STOP": "78",
        "KEY_SYSREQ": "9a",
        "KEY_UNDO": "7a",
        "KEY_UPARROW": "52",
        "KEY_VOLUMEDOWN": "81",
        "KEY_VOLUMEUP": "80",
        "KEY_YEN": "89",
        "KEY_ZENKAKUHANKAKU": "94",
        "TAB": "2b",
      },
      "language": "global",
      "modifiers": Object {
        "MOD_LALT": "04",
        "MOD_LCTRL": "01",
        "MOD_LMETA": "08",
        "MOD_LSHIFT": "02",
        "MOD_NONE": "00",
        "MOD_RALT": "40",
        "MOD_RCTRL": "10",
        "MOD_RMETA": "80",
        "MOD_RSHIFT": "20",
      },
    },
    "regions": Object {
      "en_us": Object {
        "keys": Object {
          "!": "02,1e",
          "\\"": "02,34",
          "#": "02,20",
          "$": "02,21",
          "%": "02,22",
          "&": "02,24",
          "'": "00,34",
          "(": "02,26",
          ")": "02,27",
          "*": "02,25",
          "+": "02,2e",
          ",": "00,36",
          "-": "00,2d",
          ".": "00,37",
          "/": "00,38",
          "0": "00,27",
          "1": "00,1e",
          "2": "00,1f",
          "3": "00,20",
          "4": "00,21",
          "5": "00,22",
          "6": "00,23",
          "7": "00,24",
          "8": "00,25",
          "9": "00,26",
          ":": "02,33",
          ";": "00,33",
          "<": "02,36",
          "=": "00,2e",
          ">": "02,37",
          "?": "02,38",
          "@": "02,1f",
          "A": "02,04",
          "B": "02,05",
          "C": "02,06",
          "D": "02,07",
          "E": "02,08",
          "F": "02,09",
          "G": "02,0a",
          "H": "02,0b",
          "I": "02,0c",
          "J": "02,0d",
          "K": "02,0e",
          "L": "02,0f",
          "M": "02,10",
          "N": "02,11",
          "O": "02,12",
          "P": "02,13",
          "Q": "02,14",
          "R": "02,15",
          "S": "02,16",
          "T": "02,17",
          "U": "02,18",
          "V": "02,19",
          "W": "02,1a",
          "X": "02,1b",
          "Y": "02,1c",
          "Z": "02,1d",
          "[": "00,2f",
          "\\\\": "00,31",
          "]": "00,30",
          "^": "02,23",
          "_": "02,2d",
          "\`": "00,35",
          "a": "00,04",
          "b": "00,05",
          "c": "00,06",
          "d": "00,07",
          "e": "00,08",
          "f": "00,09",
          "g": "00,0a",
          "h": "00,0b",
          "i": "00,0c",
          "j": "00,0d",
          "k": "00,0e",
          "l": "00,0f",
          "m": "00,10",
          "n": "00,11",
          "o": "00,12",
          "p": "00,13",
          "q": "00,14",
          "r": "00,15",
          "s": "00,16",
          "t": "00,17",
          "u": "00,18",
          "v": "00,19",
          "w": "00,1a",
          "x": "00,1b",
          "y": "00,1c",
          "z": "00,1d",
          "{": "02,2f",
          "|": "02,31",
          "}": "02,30",
          "~": "02,35",
        },
        "language": "en_us",
      },
      "fr": Object {
        "keys": Object {
          "!": "00,38",
          "\\"": "00,20",
          "$": "00,30",
          "%": "02,34",
          "&": "00,1e",
          "'": "00,21",
          "(": "00,22",
          ")": "00,2d",
          "*": "00,32",
          "+": "02,2e",
          ",": "00,10",
          "-": "00,23",
          ".": "02,36",
          "/": "02,37",
          "0": "02,27",
          "1": "02,1e",
          "2": "02,1f",
          "3": "02,20",
          "4": "02,21",
          "5": "02,22",
          "6": "02,23",
          "7": "02,24",
          "8": "02,25",
          "9": "02,26",
          ":": "00,37",
          ";": "00,36",
          "=": "00,2e",
          "?": "02,10",
          "A": "02,14",
          "B": "02,05",
          "C": "02,06",
          "D": "02,07",
          "E": "02,08",
          "F": "02,09",
          "G": "02,0a",
          "H": "02,0b",
          "I": "02,0c",
          "J": "02,0d",
          "K": "02,0e",
          "L": "02,0f",
          "M": "02,33",
          "N": "02,11",
          "O": "02,12",
          "P": "02,13",
          "Q": "02,04",
          "R": "02,15",
          "S": "02,16",
          "T": "02,17",
          "U": "02,18",
          "V": "02,19",
          "W": "02,1d",
          "X": "02,1b",
          "Y": "02,1c",
          "Z": "02,1a",
          "^": "00,2f",
          "_": "00,25",
          "a": "00,14",
          "b": "00,05",
          "c": "00,06",
          "d": "00,07",
          "e": "00,08",
          "f": "00,09",
          "g": "00,0a",
          "h": "00,0b",
          "i": "00,0c",
          "j": "00,0d",
          "k": "00,0e",
          "l": "00,0f",
          "m": "00,33",
          "n": "00,11",
          "o": "00,12",
          "p": "00,13",
          "q": "00,04",
          "r": "00,15",
          "s": "00,16",
          "t": "00,17",
          "u": "00,18",
          "v": "00,19",
          "w": "00,1d",
          "x": "00,1b",
          "y": "00,1c",
          "z": "00,1a",
          "£": "02,30",
          "§": "02,38",
          "¨": "02,2f",
          "°": "02,2d",
          "µ": "02,32",
          "à": "00,27",
          "ç": "00,26",
          "è": "00,24",
          "é": "00,1f",
          "ù": "00,34",
        },
        "language": "fr",
      },
    },
  },
  "scancodes": Object {
    "global": Object {
      "keys": Object {
        "28": "ENTER",
        "29": "ESC",
        "2a": "BACKSPACE",
        "2b": "TAB",
        "2c": " ",
        "39": "KEY_CAPSLOCK",
        "3a": "KEY_F1",
        "3b": "KEY_F2",
        "3c": "KEY_F3",
        "3d": "KEY_F4",
        "3e": "KEY_F5",
        "3f": "KEY_F6",
        "40": "KEY_F7",
        "41": "KEY_F8",
        "42": "KEY_F9",
        "43": "KEY_F10",
        "44": "KEY_F11",
        "45": "KEY_F12",
        "46": "KEY_PRINTSCREEN",
        "47": "KEY_SCROLLLOCK",
        "48": "KEY_PAUSE",
        "49": "KEY_INSERT",
        "4a": "KEY_HOME",
        "4b": "KEY_PAGEUP",
        "4c": "KEY_DELETE",
        "4d": "KEY_END",
        "4e": "KEY_PAGEDOWN",
        "4f": "KEY_RIGHTARROW",
        "50": "KEY_LEFTARROW",
        "51": "KEY_DOWNARROW",
        "52": "KEY_UPARROW",
        "53": "KEY_NUMLOCK",
        "54": "KEY_KPAD_SLASH",
        "55": "KEY_KPAD_ASTERISK",
        "56": "KEY_KPAD_MINUS",
        "57": "KEY_KPAD_PLUS",
        "58": "KEY_KPAD_ENTER",
        "59": "KEY_KPAD_1",
        "5a": "KEY_KPAD_2",
        "5b": "KEY_KPAD_3",
        "5c": "KEY_KPAD_4",
        "5d": "KEY_KPAD_5",
        "5e": "KEY_KPAD_6",
        "5f": "KEY_KPAD_7",
        "60": "KEY_KPAD_8",
        "61": "KEY_KPAD_9",
        "62": "KEY_KPAD_0",
        "63": "KEY_KPAD_DOT",
        "64": "KEY_102ND",
        "65": "KEY_COMPOSE",
        "66": "KEY_POWER",
        "67": "KEY_KPEQUAL",
        "68": "KEY_F13",
        "69": "KEY_F14",
        "6a": "KEY_F15",
        "6b": "KEY_F16",
        "6c": "KEY_F17",
        "6d": "KEY_F18",
        "6e": "KEY_F19",
        "6f": "KEY_F20",
        "70": "KEY_F21",
        "71": "KEY_F22",
        "72": "KEY_F23",
        "73": "KEY_F24",
        "74": "KEY_OPEN",
        "75": "KEY_HELP",
        "76": "KEY_PROPS",
        "77": "KEY_FRONT",
        "78": "KEY_STOP",
        "79": "KEY_AGAIN",
        "7a": "KEY_UNDO",
        "7b": "KEY_CUT",
        "7c": "KEY_COPY",
        "7d": "KEY_PASTE",
        "7e": "KEY_FIND",
        "7f": "KEY_MUTE",
        "80": "KEY_VOLUMEUP",
        "81": "KEY_VOLUMEDOWN",
        "82": "KEY_LOCKING_CAPSLOCK",
        "83": "KEY_LOCKING_NUMLOCK",
        "84": "KEY_LOCKING_SCROLLLOCK",
        "85": "KEY_KPAD_COMMA",
        "86": "KEY_KPAD_EQUAL",
        "87": "KEY_RO",
        "88": "KEY_KATAKANAHIRAGANA",
        "89": "KEY_YEN",
        "8a": "KEY_HENKAN",
        "8b": "KEY_MUHENKAN",
        "8c": "KEY_KPJPCOMMA",
        "8d": "KEY_INT7",
        "8e": "KEY_INT8",
        "8f": "KEY_INT9",
        "90": "KEY_HANGEUL",
        "91": "KEY_HANJA",
        "92": "KEY_KATAKANA",
        "93": "KEY_HIRAGANA",
        "94": "KEY_ZENKAKUHANKAKU",
        "95": "KEY_LANG6",
        "96": "KEY_LANG7",
        "97": "KEY_LANG8",
        "98": "KEY_LANG9",
        "99": "KEY_ALTERASE",
        "9a": "KEY_SYSREQ",
        "9b": "KEY_CANCEL",
        "9c": "KEY_CLEAR",
        "9d": "KEY_PRIOR",
        "9e": "KEY_RETURN",
        "9f": "KEY_SEPARATOR",
        "a0": "KEY_OUT",
        "a1": "KEY_OPER",
        "a2": "KEY_CLEAR",
        "a3": "KEY_CRSEL",
        "a4": "KEY_EXSEL",
        "b0": "KEY_KPAD_00",
        "b1": "KEY_KPAD_000",
        "b2": "KEY_KPAD_THOUSANDSSEPARATOR",
        "b3": "KEY_KPAD_DECIMALSEPARATOR",
        "b4": "KEY_KPAD_CURRENCYUNIT",
        "b5": "KEY_KPAD_CURRENCYSUBUNIT",
        "b6": "KEY_KPAD_LEFTPAREN",
        "b7": "KEY_KPAD_RIGHTPAREN",
        "b8": "KEY_KPAD_{",
        "b9": "KEY_KPAD_}",
        "ba": "KEY_KPAD_Tab",
        "bb": "KEY_KPAD_Backspace",
        "bc": "KEY_KPAD_A",
        "bd": "KEY_KPAD_B",
        "be": "KEY_KPAD_C",
        "bf": "KEY_KPAD_D",
        "c0": "KEY_KPAD_E",
        "c1": "KEY_KPAD_F",
        "c2": "KEY_KPAD_XOR",
        "c3": "KEY_KPAD_^",
        "c4": "KEY_KPAD_%",
        "c5": "KEY_KPAD_<",
        "c6": "KEY_KPAD_>",
        "c7": "KEY_KPAD_&",
        "c8": "KEY_KPAD_&&",
        "c9": "KEY_KPAD_|",
        "ca": "KEY_KPAD_||",
        "cb": "KEY_KPAD_:",
        "cc": "KEY_KPAD_#",
        "cd": "KEY_KPAD_Space",
        "ce": "KEY_KPAD_@",
        "cf": "KEY_KPAD_!",
        "d0": "KEY_KPAD_MEMSTORE",
        "d1": "KEY_KPAD_MEMRECALL",
        "d2": "KEY_KPAD_MEMCLEAR",
        "d3": "KEY_KPAD_MEMADD",
        "d4": "KEY_KPAD_MEMSUB",
        "d5": "KEY_KPAD_MEMMULT",
        "d6": "KEY_KPAD_MEMDIV",
        "d7": "KEY_KPAD_PLUSMINUS",
        "d8": "KEY_KPAD_CLEAR",
        "d9": "KEY_KPAD_CLEARENTRY",
        "da": "KEY_KPAD_BINARY",
        "db": "KEY_KPAD_OCTAL",
        "dc": "KEY_KPAD_DECIMAL",
        "dd": "KEY_KPAD_HEXADECIMAL",
        "e0": "KEY_LEFTCTRL",
        "e1": "KEY_LEFTSHIFT",
        "e2": "KEY_LEFTALT",
        "e3": "KEY_LEFTMETA",
        "e4": "KEY_RIGHTCTRL",
        "e5": "KEY_RIGHTSHIFT",
        "e6": "KEY_RIGHTALT",
        "e7": "KEY_RIGHTMETA",
        "e8": "KEY_MEDIA_PLAYPAUSE",
        "e9": "KEY_MEDIA_STOPCD",
        "ea": "KEY_MEDIA_PREVIOUSSONG",
        "eb": "KEY_MEDIA_NEXTSONG",
        "ec": "KEY_MEDIA_EJECTCD",
        "ed": "KEY_MEDIA_VOLUMEUP",
        "ee": "KEY_MEDIA_VOLUMEDOWN",
        "ef": "KEY_MEDIA_MUTE",
        "f0": "KEY_MEDIA_WWW",
        "f1": "KEY_MEDIA_BACK",
        "f2": "KEY_MEDIA_FORWARD",
        "f3": "KEY_MEDIA_STOP",
        "f4": "KEY_MEDIA_FIND",
        "f5": "KEY_MEDIA_SCROLLUP",
        "f6": "KEY_MEDIA_SCROLLDOWN",
        "f7": "KEY_MEDIA_EDIT",
        "f8": "KEY_MEDIA_SLEEP",
        "f9": "KEY_MEDIA_COFFEE",
        "fa": "KEY_MEDIA_REFRESH",
        "fb": "KEY_MEDIA_CALC",
      },
      "language": "global",
      "modifiers": Object {
        "00": "MOD_NONE",
        "01": "MOD_LCTRL",
        "02": "MOD_LSHIFT",
        "04": "MOD_LALT",
        "08": "MOD_LMETA",
        "10": "MOD_RCTRL",
        "20": "MOD_RSHIFT",
        "40": "MOD_RALT",
        "80": "MOD_RMETA",
      },
    },
    "regions": Object {
      "en_us": Object {
        "keys": Object {
          "04": Array [
            "a",
            "A",
          ],
          "05": Array [
            "b",
            "B",
          ],
          "06": Array [
            "c",
            "C",
          ],
          "07": Array [
            "d",
            "D",
          ],
          "08": Array [
            "e",
            "E",
          ],
          "09": Array [
            "f",
            "F",
          ],
          "0a": Array [
            "g",
            "G",
          ],
          "0b": Array [
            "h",
            "H",
          ],
          "0c": Array [
            "i",
            "I",
          ],
          "0d": Array [
            "j",
            "J",
          ],
          "0e": Array [
            "k",
            "K",
          ],
          "0f": Array [
            "l",
            "L",
          ],
          "10": Array [
            "m",
            "M",
          ],
          "11": Array [
            "n",
            "N",
          ],
          "12": Array [
            "o",
            "O",
          ],
          "13": Array [
            "p",
            "P",
          ],
          "14": Array [
            "q",
            "Q",
          ],
          "15": Array [
            "r",
            "R",
          ],
          "16": Array [
            "s",
            "S",
          ],
          "17": Array [
            "t",
            "T",
          ],
          "18": Array [
            "u",
            "U",
          ],
          "19": Array [
            "v",
            "V",
          ],
          "1a": Array [
            "w",
            "W",
          ],
          "1b": Array [
            "x",
            "X",
          ],
          "1c": Array [
            "y",
            "Y",
          ],
          "1d": Array [
            "z",
            "Z",
          ],
          "1e": Array [
            "1",
            "!",
          ],
          "1f": Array [
            "2",
            "@",
          ],
          "20": Array [
            "3",
            "#",
          ],
          "21": Array [
            "4",
            "$",
          ],
          "22": Array [
            "5",
            "%",
          ],
          "23": Array [
            "6",
            "^",
          ],
          "24": Array [
            "7",
            "&",
          ],
          "25": Array [
            "8",
            "*",
          ],
          "26": Array [
            "9",
            "(",
          ],
          "27": Array [
            "0",
            ")",
          ],
          "2d": Array [
            "-",
            "_",
          ],
          "2e": Array [
            "=",
            "+",
          ],
          "2f": Array [
            "[",
            "{",
          ],
          "30": Array [
            "]",
            "}",
          ],
          "31": Array [
            "\\\\",
            "|",
          ],
          "32": Array [
            "\`",
            "~",
          ],
          "33": Array [
            ";",
            ":",
          ],
          "34": Array [
            "'",
            "\\"",
          ],
          "35": Array [
            "\`",
            "~",
          ],
          "36": Array [
            ",",
            "<",
          ],
          "37": Array [
            ".",
            ">",
          ],
          "38": Array [
            "/",
            "?",
          ],
        },
        "language": "en_us",
      },
      "fr": Object {
        "keys": Object {
          "04": Array [
            "q",
            "Q",
          ],
          "05": Array [
            "b",
            "B",
          ],
          "06": Array [
            "c",
            "C",
          ],
          "07": Array [
            "d",
            "D",
          ],
          "08": Array [
            "e",
            "E",
          ],
          "09": Array [
            "f",
            "F",
          ],
          "0a": Array [
            "g",
            "G",
          ],
          "0b": Array [
            "h",
            "H",
          ],
          "0c": Array [
            "i",
            "I",
          ],
          "0d": Array [
            "j",
            "J",
          ],
          "0e": Array [
            "k",
            "K",
          ],
          "0f": Array [
            "l",
            "L",
          ],
          "10": Array [
            ",",
            "?",
          ],
          "11": Array [
            "n",
            "N",
          ],
          "12": Array [
            "o",
            "O",
          ],
          "13": Array [
            "p",
            "P",
          ],
          "14": Array [
            "a",
            "A",
          ],
          "15": Array [
            "r",
            "R",
          ],
          "16": Array [
            "s",
            "S",
          ],
          "17": Array [
            "t",
            "T",
          ],
          "18": Array [
            "u",
            "U",
          ],
          "19": Array [
            "v",
            "V",
          ],
          "1a": Array [
            "z",
            "Z",
          ],
          "1b": Array [
            "x",
            "X",
          ],
          "1c": Array [
            "y",
            "Y",
          ],
          "1d": Array [
            "w",
            "W",
          ],
          "1e": Array [
            "&",
            "1",
          ],
          "1f": Array [
            "é",
            "2",
          ],
          "20": Array [
            "\\"",
            "3",
          ],
          "21": Array [
            "'",
            "4",
          ],
          "22": Array [
            "(",
            "5",
          ],
          "23": Array [
            "-",
            "6",
          ],
          "24": Array [
            "è",
            "7",
          ],
          "25": Array [
            "_",
            "8",
          ],
          "26": Array [
            "ç",
            "9",
          ],
          "27": Array [
            "à",
            "0",
          ],
          "2d": Array [
            ")",
            "°",
          ],
          "2e": Array [
            "=",
            "+",
          ],
          "2f": Array [
            "^",
            "¨",
          ],
          "30": Array [
            "$",
            "£",
          ],
          "32": Array [
            "*",
            "µ",
          ],
          "33": Array [
            "m",
            "M",
          ],
          "34": Array [
            "ù",
            "%",
          ],
          "36": Array [
            ";",
            ".",
          ],
          "37": Array [
            ":",
            "/",
          ],
          "38": Array [
            "!",
            "§",
          ],
        },
        "language": "fr",
      },
    },
  },
}
`;
